fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java

```diff
@@ -225,4 +229,40 @@ public Response getResource(@HeaderParam("Range") final String rangeValue)
             readLock.release();
         }
     }
+
+    /**
+     * Deletes an object.
+     *
+     * @return response
+     */
+    @DELETE
+    @Timed
+    public Response deleteObject() throws ItemNotFoundException {
+
+        hasRestrictedPath(externalPath);
+        LOGGER.info("Delete resource '{}'", externalPath);
+
+        final AcquiredLock lock = lockManager.lockForDelete(resource().getPath());
+
+        try {
+            final FedoraResource aclResource = resource().getAcl();
+            if (aclResource != null) {
+                aclResource.delete();
+            }
+            session.commit();
+
+            if (aclResource == null) {
+                throwItemNotFoundException();
+            }
+
+            return noContent().build();
+
+        } finally {
+            lock.release();
+        }
+    }
+
+    private void throwItemNotFoundException() throws ItemNotFoundException {
+        throw new ItemNotFoundException("Resource does not exist.");
```

Comment:
I am happy to move this through as-is... but is it helpful to wrap the one-line throwing of an exception in a method?

Reply:
A valid point.  It looks like the message isn't being passed on to the response anyway by the ItemNotFoundExceptionMapper.  Fixed.